medio/backends/itk_io.py

```diff
@@ -43,13 +44,28 @@ def read_img(
         :return: numpy image and metadata object which includes pixdim, affine, original orientation string and
         coordinates system
         """
+        if header:
+            imageio = itk.GDCMImageIO.New()
+            if private_tags:
+                imageio.LoadPrivateTagsOn()
+            # we need to set the imageio to the reader, but with fallback_only=True it will not set it unless it fails
+            fallback_only = False
+        else:
+            imageio = None
         input_path = Path(input_path)
         if input_path.is_dir():
             img = ItkIO.read_dir(
-                str(input_path), pixel_type, fallback_only, series=series, header=header
+                str(input_path), pixel_type, fallback_only, series, imageio
             )
         elif input_path.is_file():
-            img = ItkIO.read_img_file(str(input_path), pixel_type, fallback_only)
+            # If the input is not a dicom, fallback to not use imageio.
+            # NOTE: this assume imageio is used for dicom files only. If it changed, we need to modify this.
+            use_dicom_imageio = imageio and imageio.CanReadFile(str(input_path))
+            if not use_dicom_imageio:
+                imageio = None
```

Comment:
I suggest moving this logic above, to the `if header:` block.
Otherwise, `fallback_only` can be set to `False` (L52) even when the image isn't a dicom (`use_dicom_imageio` is `False`).

Alternatively, store the original value of the given `fallback_only` and reset `fallback_only` to its original value in the `if not use_dicom_imageio:` block.

Reply:
I need to check it only in case it's a file. For a folder it won't work. 
But for a folder I can assume it's always a DICOM, right? 
See the new code

Reply:
> But for a folder I can assume it's always a DICOM, right?

Yes